src/Tribe/Body_Classes.php

```diff
@@ -0,0 +1,223 @@
+<?php
+/**
+ * Class used to manage and body classes across our plugins.
+ *
+ * @since TBD
+ */
+namespace Tribe;
+
+if ( ! class_exists( 'Body_Classes' ) ) {
+	class Body_Classes {
+		/**
+		 * Stores all the classes.
+		 * In the format: ['class' => true, 'class => false ]
+		 *
+		 * @var array
+		 */
+		protected $classes = [];
+
+		/**
+		 * Static singleton factory method.
+		 *
+		 * @since TBD
+		 *
+		 * @return self
+		 */
+		public static function instance() {
+			return tribe( self::class );
+		}
+
+		/**
+		 * Register our methods in the correct places.
+		 *
+		 * @since TBD
+		 */
+		public function __construct() {
+			// Hook the actual adding of body classes.
+			add_action( 'body_class', [ $this, 'maybe_add_body_classes' ] );
```

Comment:
Do not hook in the constructor.
Constructors should not have side-effects and should only prepare the object to be used.
Remove this, along w/ the `instance` method, and use the container/service-locator correctly.

Somewhere in a service provider or Main:

```php
tribe_singleton('body-classes', Body_Classes::class);
add_action( 'body_class', [ $this, 'add_body_classes' ] );
```

```php
public function add_body_classes($classes = []){
  $body_classes = tribe('body-classes');

  return $body_classes->add_body_classes($classes);
}
```